### PR TITLE
Migrates some test scenarios from checkout_spec.rb

### LINKED ISSUE #8670: Test coverage on the "1 - Your details" section
### What we should change and why (this is tech debt)

Relates to #8667.

### Context

This issue accounts for test coverage on the `/checkout/details` page and its functionality.

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

Partially addresses #8670.

Introduces a separate testing helper for split-checkout.
Adapts existing system tests from `checkout_spec.rb` to test the split-checkout feature.

#### What should we test?

Green build.

#### Release notes

Changelog Category: Technical changes

#### Documentation updates

Some tests here will go red, depending on the status of this PR: https://github.com/openfoodfoundation/openfoodnetwork/issues/8668

### CODE REVIEW

**Comment by filipefurtad0** [comment]
There are still some missing scenarios here, like:
- mandatory fields are required
- existing customer address is filed in when visiting this section
- saving default addresses

So even if we merge this, the issue is not yet closed. Another option is to add these to this PR.

**Review by achauve — APPROVED** [review]

File: spec/support/request/split_checkout_request_helper.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module SplitCheckoutRequestsHelper
+  def have_split_checkout_details
+    have_content "Your details"
+  end
+
+  def split_checkout_as_guest
```
Comment (Matt-Yorkley) [inline spec/support/request/split_checkout_request_helper.rb:8]: If we've created a new/separate class (`SplitCheckoutRequestsHelper`) and included it in the spec, we don't really need to preface all these methods with `split_`. `checkout_as_guest` is a lot nicer to read than `split_checkout_as_guest`.

File: spec/support/request/split_checkout_request_helper.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module SplitCheckoutRequestsHelper
```
Comment (Matt-Yorkley) [inline spec/support/request/split_checkout_request_helper.rb:3]: This isn't a request helper and probably shouldn't live in `/support/request/`; it's methods are checking things in the DOM (which doesn't happen in request specs) and it's being used for system tests.

Comment (Matt-Yorkley) [reply spec/support/request/split_checkout_request_helper.rb:3]: Ah I see the previous checkout helper is named like that. Well... it's wrong.

**Naming things correctly** in software development is both notoriously hard and deceptively important :sweat_smile:

File: spec/system/consumer/split_checkout_spec.rb [hunk 3]
```diff
@@ -85,4 +86,64 @@
...
+        end
+      end
+    end
+  end
+
+  context "when I have an out of stock product in my cart" do
+    before do
+      variant.on_demand = false
+      variant.on_hand = 0
+      variant.save!
```
Comment (Matt-Yorkley) [inline spec/system/consumer/split_checkout_spec.rb:138]: These three lines could probably just be one; `variant.update!(on_demand: false, on_hand: 0)`.

File: spec/support/request/split_checkout_request_helper.rb (hunk 2 above)
Comment (filipefurtad0) [reply spec/support/request/split_checkout_request_helper.rb]: Great catch, thank you :+1:

**Review by Matt-Yorkley — APPROVED** [review]
Great :muscle:

File: spec/system/consumer/split_checkout_spec.rb [hunk 4]
```diff
@@ -85,4 +86,64 @@
...
+      login_as(user)
+      visit checkout_path
+    end
+
+    describe "purchasing" do
+      context "takes us to the Payment Method" do
+        before do
+          split_fill_out_details
+          split_fill_out_billing_address
+        end
```
[the remark this answers is not in the document]
Comment (filipefurtad0) [reply spec/system/consumer/split_checkout_spec.rb]: This makes total sense, thanks so much for pointing this out :pray:

**Pull request merged**